cpp/oneapi/dal/algo/knn/backend/gpu/train_kernel_kd_tree_dpc.cpp

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#define DAAL_SYCL_INTERFACE
+
+#include "oneapi/dal/algo/knn/backend/gpu/train_kernel.hpp"
+#include "oneapi/dal/backend/dispatcher_dpc.hpp"
+#include "oneapi/dal/backend/interop/common_dpc.hpp"
+#include "oneapi/dal/backend/interop/error_converter.hpp"
+
+namespace oneapi::dal::knn::backend {
+
+using dal::backend::context_gpu;
+
+template <typename Float>
+struct train_kernel_gpu<Float, method::kd_tree> {
+    train_result operator()(const context_gpu& ctx,
+                            const descriptor_base& desc,
+                            const train_input& input) const {
+        throw unimplemented_error("kNN K-D Tree method is implemented for CPU only!");
```

Comment:
You only instantiate kd_tree method. It will fall apart during compilation. Right?

Reply:
All ones are instantiated in train_opsXXX